Docker image should include the `git` binary for vendir

I'm trying to use `vendir` from the docker image with this config file:
```yaml
---
apiVersion: vendir.k14s.io/v1alpha1
kind: Config
directories:
  - path: alloyModules
    contents:
      - path: .
        git:
          url: https://github.com/grafana/alloy-modules
          ref: origin/main
```

But, it fails because `git` is not in the container image:

```shell
docker run --rm --volume "$(pwd):/src" --workdir /src ghcr.io/carvel-dev/docker-image:latest vendir sync -l
apiVersion: vendir.k14s.io/v1alpha1
directories:
- contents:
  - git:
      ref: efc635083892036dd0cf2dad76a26d9ff716858f
      url: https://github.com/grafana/alloy-modules
    path: .
  path: alloyModules
kind: Config
minimumRequiredVersion: ""

vendir: Error: Syncing directory 'alloyModules':
  Syncing directory '.' with git contents:
    Fetching git repository:
      Git [init]: exec: "git": executable file not found in $PATH (stderr: )
make: *** [alloyModules/LICENSE] Error 1
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,7 @@
 FROM ubuntu:24.04
 
 # perl is needed for shasum
-RUN apt-get -y update && apt-get install -y wget perl ca-certificates && update-ca-certificates && rm -rf /var/lib/apt/lists/*
+RUN apt-get -y update && apt-get install -y git wget perl ca-certificates && update-ca-certificates && rm -rf /var/lib/apt/lists/*
 
 RUN bash -c "set -eo pipefail; wget -O- https://carvel.dev/install.sh | bash"
 

